catalog/app/containers/NavBar/Search.tsx

```diff
@@ -0,0 +1,362 @@
+    overflow: 'hidden',
+    padding: `0 ${t.spacing(1)}px 0 0`,
+    position: 'absolute',
+    right: 0,
+    transition: [
+      `background-color ${expandAnimationDuration}ms`,
+      `opacity ${expandAnimationDuration}ms`,
+      `width ${expandAnimationDuration}ms`,
+    ].join(', '),
+    width: t.spacing(24),
+    '&:not($iconized)': {
+      background: fade(t.palette.common.white, 0.1),
+    },
+    '&:not($disabled):not($iconized):hover': {
+      background: fade(t.palette.common.white, 0.2),
+    },
+  },
+  iconized: {
+    width: t.spacing(4),
+  },
+  disabled: {
+    opacity: 0.8,
+  },
+  hidden: {
+    opacity: 0,
+  },
+  expanded: {
+    background: `${fade(t.palette.common.white, 0.2)} !important`,
+    width: '100%',
+  },
+  input: {
+    paddingLeft: t.spacing(1),
+    paddingTop: 8,
+    paddingBottom: 9,
+    textOverflow: 'ellipsis',
+    transition: 'opacity 200ms',
+    '$iconized:not($expanded) &': {
+      opacity: 0,
+    },
+  },
+  inputIcon: {
+    cursor: 'pointer',
+    '$root$disabled &': {
+      cursor: 'not-allowed',
+    },
+  },
+  inputOptions: {
+    borderRadius: 0,
+    borderWidth: '0 1px 0 0',
+    paddingLeft: t.spacing(1.25),
+    paddingRight: t.spacing(0.25),
+  },
+}))
+
+interface SearchInputExternalProps {
+  hidden?: boolean
+  iconized?: boolean
+}
+
+type SearchInputBaseProps = Pick<
+  M.InputBaseProps,
+  'disabled' | 'value' | 'placeholder' | 'onFocus' | 'onKeyDown' | 'onChange'
+>
+
+interface SearchInputProps extends SearchInputExternalProps, SearchInputBaseProps {
+  expanded?: boolean
+  helpOpen?: boolean
+  onHelpToggle?: () => void
+}
+
+function SearchInput({
+  expanded = false,
+  helpOpen = false,
+  hidden = false,
+  iconized = false,
+  onHelpToggle,
+  ...props
+}: SearchInputProps) {
+  const classes = useInputStyles()
+  return (
+    <M.InputBase
+      classes={{ root: classes.root, input: classes.input }}
+      className={cx({
+        [classes.expanded]: expanded,
+        [classes.disabled]: props.disabled,
+        [classes.iconized]: iconized,
+        [classes.hidden]: hidden,
+      })}
+      {...props}
+      startAdornment={
+        <M.InputAdornment position="start">
+          {iconized && !expanded ? (
+            <M.Icon className={classes.inputIcon} onClick={onHelpToggle}>
+              search
+            </M.Icon>
+          ) : (
+            <Lab.ToggleButton
+              className={classes.inputOptions}
+              size="small"
+              value="help"
+              selected={helpOpen}
+              onChange={onHelpToggle}
+              disabled={props.disabled}
+            >
+              <M.Icon fontSize="small">search</M.Icon>
+              <M.Icon fontSize="small">
+                {helpOpen ? 'arrow_drop_up' : 'arrow_drop_down'}
+              </M.Icon>
+            </Lab.ToggleButton>
+          )}
+        </M.InputAdornment>
+      }
+    />
+  )
+}
+
+function useSearchUrlState(bucket: string | undefined) {
```

Comment:
```suggestion
function useSearchUrlState(bucket?: string) {
```

Reply:
that actually means something different, but in this case it would work as well